### ISSUE TITLE
Get builds going on packet.net

### ISSUE THREAD

**Issue by gerkey** [ros2/ci#46]
Notes to self:

~~~
root@ros2-testing:~/code/ci# history
    3  apt-get update
    4  apt-get install docker.io
    6  mkdir code
    7  cd code
    8  git clone https://github.com/ros2/ci
    9  cd ci
   12  git checkout packet_aarch64
   20  service docker start
   21  docker build --build-arg PLATFORM=arm -t ros2_batch_ci_aarch64 linux_docker_resources
   23  mkdir ~/.ccache
   26  docker run --privileged -e UID=1001 -e GID=1001 -e CI_ARGS="--do-venv --force-ansi-color --workspace-path /root/code/ci --fastrtps --isolated --cmake-build-type Debug --ament-build-args --parallel -- --ament-test-args --ctest-args -LE linter --" -e CCACHE_DIR=/home/rosbuild/.ccache -i -v /root/code/ci:/home/rosbuild/ci_scripts -v /root/.ccache:/home/rosbuild/.ccache ros2_batch_ci_aarch64
~~~

**Comment by gerkey** [comment]
Summary so far:

* With a few changes (https://github.com/ros2/ci/commit/9e3afcb13f1649cee5dbdadbba2fbfc228505928), the docker-based build-and-test job seems to work, and there were only a handful of test failures.
* I'm running it again with the usual args (`--retest-until-pass 10`, `--parallel`, etc.) and with a warm ccache to get a sense for performance.

**Comment by gerkey** [comment]
With a warm ccache, the standard CI build and test took 1h16m and produced 3 test failures. Not too shabby.

~~~
$ time docker run --privileged -e UID=1001 -e GID=1001 -e CI_ARGS="--do-venv --force-ansi-color --workspace-path /root/code/ci --fastrtps --isolated --cmake-build-type Debug --ament-build-args --parallel -- --ament-test-args --retest-until-pass 10 --" -e CCACHE_DIR=/home/rosbuild/.ccache -i -v /root/code/ci:/home/rosbuild/ci_scripts -v /root/.ccache:/home/rosbuild/.ccache ros2_batch_ci_aarch64
real	76m30.242s
user	0m5.504s
sys	0m2.636s
~~~

~~~
$ ./ws/src/ament/ament_tools/scripts/ament.py test_results
Skipping 'ws/src/ros2/robot_model/kdl_parser/test/pr2_desc_vector.xml': File "<string>", line None, xml.etree.ElementTree.ParseError: mismatched tag: line 22, column 2
Skipping 'ws/src/ros2/robot_model/kdl_parser/test/pr2_desc_bracket.xml': File "<string>", line None, xml.etree.ElementTree.ParseError: mismatched tag: line 21, column 2
Skipping 'ws/src/ros2/robot_model/kdl_parser/test/pr2_desc_bracket2.xml': File "<string>", line None, xml.etree.ElementTree.ParseError: mismatched tag: line 21, column 2
ws/build/rosidl_generator_c/test_results/rosidl_generator_c/test_interfaces_c.xml: 1 tests, 0 skipped, 0 errors, 1 failures
ws/build/test_communication/test_results/test_communication/test_requester_replier__Empty__rclcpp__rclpy__rmw_fastrtps_cpp.xunit.xml: 1 tests, 0 skipped, 0 errors, 1 failures
ws/build/test_rclcpp/test_results/test_rclcpp/gtest_multithreaded__rmw_fastrtps_cpp.gtest.xml: 1 tests, 0 skipped, 0 errors, 1 failures
Summary: 8382 tests, 0 errors, 3 failures, 4 skipped
~~~

**Comment by mikaelarguedas** [comment]
that's awesome ! :tada: 
Some of these tests are known for their flakiness so it's not very surprising.

**Comment by gerkey** [comment]
Here's the first real job running on the farm:

http://ci.ros2.org/job/ci_linux-aarch64/27

(That's a cold cache, because it's running as `jenkins`, instead of `root`.)

**Comment by wjwwood** [comment]
@gerkey what about the nightly job? should it also be made to run on the `linux_aarch64` label?

**Comment by wjwwood** [comment]
Since we're down a Linux machine again (slave on master), I went ahead and changed the aarch64 nightly to run on packet.net, and canceled the job that was blocking the rest of the nightly queue. I figure this is safe to do since the packet.net machine is up and running and because we have some nightly history on the qemu nightlies already.

**Comment by wjwwood** [comment]
actually that didn't seem to work so I changed it back to the normal label and requeued the job:

http://ci.ros2.org/job/nightly_linux-aarch64_debug/37/ (this is the one that failed)

**Comment by gerkey** [comment]
Definitely we'll want to move the CI and the nightly job. First I wanted to check that it would work, and http://ci.ros2.org/job/ci_linux-aarch64/27/ gives a good result. Now I need to tweak and merge these changes: https://github.com/ros2/ci/compare/packet_aarch64 (your build probably failed because you weren't pulling from that branch).

**Comment by gerkey** [comment]
Fixed by #47.
